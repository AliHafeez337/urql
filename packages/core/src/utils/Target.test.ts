import { Target } from './Target';

describe('on construct', () => {
  it('matches snapshot', () => {
    expect(new Target()).toMatchSnapshot();
  });
});

describe('on addEventListener -> dispatchEvent', () => {
  it('calls event listeners with event', () => {
    const t = new Target();
    const listeners = [jest.fn(), jest.fn(), jest.fn()];
    const event = { type: 'DEBUG_TEST' } as any;

    listeners.forEach(t.addEventListener);
    t.dispatchEvent(event);

    listeners.forEach(l => {
      expect(l).toBeCalledTimes(1);
      expect(l).toBeCalledWith(event);
    });
  });
});

describe('on removeEventListener -> dispatchEvent', () => {
  it('does not call removed event listeners', () => {
    const t = new Target();
    const listeners = [jest.fn(), jest.fn(), jest.fn()];
    const excludedListener = jest.fn();
    const event = { type: 'DEBUG_TEST' } as any;

    t.addEventListener(excludedListener);
    listeners.forEach(t.addEventListener);
    t.removeEventListener(excludedListener);
    t.dispatchEvent(event);

    expect(excludedListener).toBeCalledTimes(0);
    listeners.forEach(l => {
      expect(l).toBeCalledTimes(1);
    });
  });
});
