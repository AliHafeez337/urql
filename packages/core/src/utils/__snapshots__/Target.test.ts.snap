// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on construct matches snapshot 1`] = `
Target {
  "addEventListener": [Function],
  "dispatchEvent": [Function],
  "listeners": Array [],
  "removeEventListener": [Function],
}
`;
